hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/GrpcContainerUploader.java

```diff
@@ -60,8 +61,9 @@ public OutputStream startUpload(long containerId, DatanodeDetails target,
       StreamObserver<SendContainerRequest> requestStream = client.upload(
           new SendContainerResponseStreamObserver(containerId, target,
               callback));
-      return new SendContainerOutputStream(requestStream, containerId,
-          GrpcReplicationService.BUFFER_SIZE, compression) {
+      return new SendContainerOutputStream(
+          (CallStreamObserver<SendContainerRequest>) requestStream,
```

Comment:
Why not change ```private static class SendContainerResponseStreamObserver
      implements StreamObserver<SendContainerResponse> {``` to be `CallStreamObserver` 
The casting seems odd.

Reply:
actually, we don't take the `SendContainerResponseStreamObserver` as the input for SendContainerOutputStream, but the StreamObserver given by gRPC. And `StreamObserver` is the type gRPC gives. Internally, it returns an instance that implements `CallStreamObserver`, the reduces the result type to `StreamObserver`

Reply:
Please add a few lines of comments explaining the casts that are done.

Reply:
Done.